src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -4069,8 +4076,21 @@ internal uint GetValEscape(BoundExpression expr, uint scopeOfTheContainingExpres
                             isRefEscape: false);
                     }
 
-                    return Math.Max(GetValEscape(binary.Left, scopeOfTheContainingExpression),
-                                    GetValEscape(binary.Right, scopeOfTheContainingExpression));
+#if DEBUG
+                    // Nested binary operators on the left side are not visited by BoundTreeWalkerWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator.
+                    var assertNestedVisited = binary.Left is not BoundBinaryOperator;
```

Comment:
>var assertNestedVisited = binary.Left is not BoundBinaryOperator; [](http://example.com/codeflow?start=20&length=65)

It is probably better to mark the node as visited right here, instead of dealing with an extra debug only parameter.

Reply:
I'm not sure how exactly would we do that - for expression like A+B+C+D calling Visit at this point on the left-hand side would recursively visit A+B+C, then we would get to this point again for A+B+C and call Visit on A+B - so we would visit the same nodes multiple times which we probably don't want.

Reply:
> I'm not sure how exactly would we do that

The assert checks that the node is in some set. Perhaps we can add an entry in the into the set right here.

Reply:
You're right, that looks like a good solution, thanks!